File: listas/python/teste_lista_duplamente_encadeada.py
from doubly_linked_list import DoublyLinkedList

lst = DoublyLinkedList()

# 1 - Add at start
lst.add_at_start(2)
lst.add_at_start(9)
print(lst.to_array())

lst.clear()

# 2 - Add at end
lst.add_at_end(2)
lst.add_at_end(4)
lst.add_at_end(8)
lst.add_at_end(32)
lst.add_at_end(128)
print(lst.to_array())

# 3 - Get the size of the list
print(f'Tamanho da lista igual a: {lst.get_size()}')

# 4 - Add at position
lst.add_at_position(3, 16)
print("Adicionado 16 à posição 3:")
print(lst.to_array())

lst.add_at_position(5, 64)
print("Adicionado 64 à posição 5:")
print(lst.to_array())

# 5 - Remove head
lst.remove_head()
print("Removido primeiro elemento:")
print(lst.to_array())

# 6 - Remove tail
lst.remove_tail()
print("Removido último elemento:")
print(lst.to_array())

# 7 - Remove at position
lst.remove_at_position(4)
print("Removido elemento da posição 4:")
print(lst.to_array())

# 8 - Remove
lst.remove(32)
print("Removido elemento 32:")
print(lst.to_array())
lst.remove(4)
print("Removido elemento 4:")
print(lst.to_array())

# 9 - Reverse list
lst.reverse()
print("Lista reversa:")
print(lst.to_array())

File: listas/javascript/teste-lista-duplamente-encadeada.js
import DoublyLinkedList from "./doubly-linked-list.js";

const list = new DoublyLinkedList();

// 1 - Add at start
list.addAtStart(2);
list.addAtStart(9);
console.log(list.toArray());

list.clear();

// 2 - Add at end
list.addAtEnd(2);
list.addAtEnd(4);
list.addAtEnd(8);
list.addAtEnd(32);
list.addAtEnd(128);
console.log(list.toArray());

// 3 - Obter o tamanho da lista
console.log(`Tamanho da lista igual a: ${list.getSize()}`);

// 4 - Add at position
list.addAtPosition(3, 16);
console.log("Adicionado 16 à posição 3:");
console.log(list.toArray());

list.addAtPosition(5, 64);
console.log("Adicionado 64 à posição 5:");
console.log(list.toArray());

// 5 - Remove head
list.removeHead();
console.log("Removido primeiro elemento:");
console.log(list.toArray());

// 6 - Remove tail
list.removeTail();
console.log("Removido último elemento:");
console.log(list.toArray());

// 7 - Remove at position
list.removeAtPosition(4);
console.log("Removido elemento da posição 4:");
console.log(list.toArray());

// 8 - Remove
list.remove(32);
console.log("Removido elemento 32:");
console.log(list.toArray());
list.remove(4);
console.log("Removido elemento 4:");
console.log(list.toArray());

// 9 - Reverse list
list.reverse();
console.log("Lista reversa:");
console.log(list.toArray());
